src/components/Personalization/index.js

```diff
@@ -44,35 +49,51 @@ const createCollect = eventManager => {
   };
 };
 
+const validateEvent = (event, logger) => {
+  const warnings = event.validate();
+
+  if (warnings.length) {
+    logger.warn(
+      `Invalid getDecisions command options:\n\t - ${warnings.join(
+        "\n\t - "
+      )}\nFor documentation covering the getDecisions command see: ${CONFIG_DOC_URI}`
+    );
+  }
+};
+
 const createPersonalization = ({ config, logger, eventManager }) => {
   const { prehidingStyle } = config;
   const authoringModeEnabled = isAuthoringMode();
   const collect = createCollect(eventManager);
   const storage = [];
   const store = value => storage.push(value);
   const ruleComponentModules = initRuleComponentModules(collect, store);
-  const disablePersonalization = payload => {
-    payload.mergeConfigOverrides({ personalization: { enabled: false } });
-  };
 
   return {
     lifecycle: {
-      onBeforeEvent({ event, isViewStart, payload }) {
+      onBeforeEvent({ event, isViewStart, scopes }) {
         if (authoringModeEnabled) {
           logger.warn("Rendering is disabled, authoring mode.");
-          disablePersonalization(payload);
+
+          mergeQuery(event, { enabled: false });
           return;
         }
 
-        if (!isViewStart) {
-          // If NOT isViewStart disable personalization
-          disablePersonalization(payload);
-        } else {
+        if (isViewStart) {
           event.expectResponse();
 
           // For viewStart we try to hide the personalization containers
           hideContainers(prehidingStyle);
+          return;
+        }
+
+        if (scopes) {
+          event.expectResponse();
+          mergeQuery(event, { scopes });
```

Comment:
If `scopes` will only ever be added to the event when the `getDecisions` command is called, I would recommend running:

```
if (scopes) {
  event.expectResponse();
  mergeQuery(event, { scopes });
}
```

from within the `getDecisions` command down below rather than doing it here and sending the `scopes` to all the components through `lifecycle.onBeforeEvent`. If other components need access to `scopes` for some reason then maybe it still makes sense to send them through `lifecycle.onBeforeEvent`, but I don't know what that use case would be.

Reply:
yes, you are right, thank you

Reply:
Just to not confuse anyone, my comment here is no longer applicable since scopes will now be passed through the `event` command.